helper/user_object.js

```diff
@@ -0,0 +1,107 @@
+// provides factory to create new users
+
+// sharedWith?: friend's uuid = boolean
+class UserDocumentMetadata {
+  constructor(isPublic, sharedWith) {
+    this.userDocObject = {
+      isPublic,
+      sharedWidth: Object.assign({}, sharedWith),
+    };
+  }
+
+  get getUserDocObject() {
```

Comment:
You are doing the getter and setter wrong. We do the same thing at work. You should follow this convention. https://www.typescriptlang.org/docs/handbook/classes.html